src/ca65/ea65.c

```diff
@@ -106,7 +106,10 @@ void GetEA (EffAddr* A)
         if (CurTok.Tok == TOK_COMMA) {
             /* [dir],y */
             NextTok ();
-            Consume (TOK_Y, "`Y' expected");
+            if (GetCPU() == CPU_C39_NATIVE)
+              Consume (TOK_X, "`X' expected");
+            else
+              Consume (TOK_Y, "`Y' expected");
             A->AddrModeSet = AM65_DIR_IND_LONG_Y;
```

Comment:
The bracket addressing modes (`AM65_*LONG*`) are used in 65816 code.  They aren't used in the code for any of the eight-bit 65xx processors.  So, you don't need to patch that section of GetEA().